src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.cpp

```diff
@@ -150,6 +150,92 @@ namespace Util
         addShapeToLayer(layer, new_shape);
     }
 
+    void VisualizerMessenger::drawTestLayer()
```

Comment:
Hmm, this looks like something that probably belongs in a ROSTest instead of in a function on the `VisualizerMessenger`? What was the reasoning behind putting it here?

Reply:
Yea I thought about a ROSTest node, but doesn't that run in the CI? Is there a way to test by having a loop in the main of the test to continuously call these VisMsgr draw methods and don't stop until it is told to?

Reply:
Yup, the ROSTest node does run in CI. What's the trouble with that?

I think the test would look something like:
```
drawCircle()
drawSquare()
drawEllipse()
wait a bit...
assert that the values we received on the ROS topic are correct
```
If that makes sense? Checkout `logger_test.cpp` for an example of how we tested the logger in a similar fashion. Feel free to let me know if this is still confusing though, and I can knock up a rough version!

Reply:
Because we don't want to automatically check if it's working. We want to qualitatively check, on the visualizer client that the shapes are drawing correctly. It's also to stress test the client as I could make this animated in a loop.

Reply:
Ah, I see now. Would it be reasonable to create two tests, one to check if it's working properly (quantitatively) that would run in CI, and one that just draws the shapes (with stress testing functionality)? 

This would entail one ROSTest, and a separate `VisualizerTester` node that we could just run normally, probably with some params to dictate what we want it to do. This would also make it easy to run different tests, as we could just have a few launch files that launch the `VisualizerTester` and do different things.

Reply:
I believe the purpose of this test is just to see if shapes are drawn properly in the Visualizer, not to validate that messages are published correctly / with the expected parameters. Therefore I still don't think this should be broken into a ROStest component? Unless we are planning to validate the construction of shape messages as part of this issue/PR.

How would we quantitatively check that shapes are drawn correctly?

But regardless I agree this code shouldn't be a member function of the `VisualierMessenger`. These calls should be made by some test code in a test node. You can basically just move this function.

Reply:
Sounds good to me. I've opened #349 to track adding these tests for the visualizer messenger.